### ISSUE TITLE
spacefn stuck

### ISSUE THREAD

**Issue by rbirkelbach** [blahlicus/arbites-family#19]
Hello,

I commented in my issue that was closed (https://github.com/blahlicus/arbites-family/issues/17), but I guess you didn't notice. The behavior (getting stuck in a certain layer) still occurs, even after reuploading the firmware which I got from the homepage and when uploading a keymap that uses space||fn instead of spacefn which I included using arbites 2.5.0003

This behavior is really annoying and makes it unbearably difficult to use the diverge tm 1 -- imagine trying to type in a password where you cannot see the symbols and the input being unpredictable. You find the keymap in the other post. Maybe you have some beta firmware I could test and see whether this still occurs which I could preferably upload via the new arbites version? Or do you think this might be a hardware fault? 

Best and many thanks,
Robert

**Comment by blahlicus** [comment]
Hey, are you sure the layout from #17 is correct? I tested it out in both releases (2.3.1 from the official site and 2.6.x from github) of my firmware on Diverge TM 1s and Diverge 3s and cannot replicate it.

The 2.3.1 should absolutely work fine without any problems whilst the 2.6.x has some debouncing problems totally unrelated to this issue so if you've got either version then it should still work with FN keys, you mentioned that you've tried updating the firmware, have you updated it on both sides of your Diverge TM?

**Comment by rbirkelbach** [comment]
This is the layout I have been using. I further modified it a bit and tried out the space||fn keys. But it did not help. The issue is mostly there when I quickly type in a couple of numbers. I did upload the firmware from the homepage to both sides and carefully checked that it's the correct side  so that master/slave doesn't get mixed up. Sometimes it helps to press the spacefn keys again or the delete key, but not always. It's happening in about one quarter of the cases I type in a sequence of numbers.

**Comment by rbirkelbach** [comment]
I'll attach the newer layout file tomorrow, as I currently don't have access to it. But the error in question is the same as I had when using the layout I uploaded to issue #17 .

What I do is I press Space||FN1 or Space||FN2 or the deprecated version of the two keys (I am not sure about Space||fn3 Space||fn4 etc) and try to type in whatever is on the layer. Mostly it's numbers or the shifted versions of the numbers, e.g. the signs !@#$^&*()-_  and I think that it happens mostly when I type in sequences, e.g 6789.  I will try to reproduce the error tomorrow by just typing in one number instead of a sequence.

In about 3 out of 4 times it works as expected and when I release the key it goes back to the primary layer. But in roughly 1 out of 4 times it acts like if it were a toggle. I have not yet really figured out why it happens only sometimes. 

I was able to reproduce it on a different physical key and also on the left half. This error is independent of typing speed. It happens when I type fast, but also when I type slow. However I have the impression it happens more often if I type fast. 

My apologies about posting it to the wrong repo. I initially thought there might be a mismatch between what is written on the arbites keys and the underlying functionality. Also apologies about being so vague, but it's really difficult to describe for me.

I was purchaser nr. 710 on July 8th 2016.
